clientd3d/dialog.c

```diff
@@ -663,6 +674,41 @@ void DisplayDescription(object_node *obj, BYTE flags, char *description,
 }
 /************************************************************************/
 /*
+* GetRaritySuffix:  Provide a rarity suffix based on the given rarity type.
+*/
+std::string GetRaritySuffix(int rarity) 
+{
+	std::string suffix;
+	
+	switch(rarity)
+	{
+        case ITEM_RARITY_GRADE_NORMAL:
+            suffix = "";
+			break;
+        case ITEM_RARITY_GRADE_MAGIC:
+            return GetString(hInst, IDS_RARITY_GRADE_MAGIC);
```

Comment:
Don't repeat all these calls to GetString.  Instead set a variable to the string to fetch, then call GetString once after the switch.

Reply:
Changed `GetRaritySuffix` to only have one call to `GetString` based on a new `StringID` variable determined by a switch case.